### PR TITLE
updated README to include newly defined syntax and EBNF

### PR DESCRIPTION
**Pull request by WillCZhang**

### CODE REVIEW

**Comment by micromoon1997** [comment]
LGTM

File: README.md [hunk 1]
```diff
@@ -71,14 +59,15 @@ specify layouts inside a layout.
...
+EBNF
+    PROGRAM     ::= (ASSIGNMENT)* (LAYOUT)+
+    ASSIGNMENT  ::= VAR “=” VALUE
+    VAR         ::= CHAR(CHAR | NUMBER | ” _”)*
+    STRING      ::= \” CHAR(CHAR)* \”
+    CHAR        ::= “a”-”z” | “A”-”Z”
+    VALUE       ::= STRING | NUMBER | CONSTRUCTOR | VAR
+    CONSTRUCTOR ::= FUNCNAME | FUNCNAME “(” ASSIGNMENT (, ASSIGNMENT)*  “)”
+    FUNCNAME    ::= “Nav” | “Header” | “Content” | “Link” | “Image” | “Video” | “Footer” | “Button”
+    LAYOUT      ::= (VAR | “Page”) “{“ ROW(“\n” ROW)* “}”
```
Comment (Raymond-yao) [inline README.md]: This should be:
LAYOUT      ::= (VAR | “Page”) “{“ ROW(ROW)* “}”
ROW         ::= "[" (VAR | CONSTRUCTOR)  (“\s” (VAR | CONSTRUCTOR))* "]"

or 
LAYOUT      ::= (VAR | “Page”) “{“ "[" ROW "]" ( "[" ROW "]" )* “}”
ROW         ::=  (VAR | CONSTRUCTOR)  (“\s” (VAR | CONSTRUCTOR))*

Comment (WillCZhang) [reply README.md]: This was copied directly from `./dsl/parser/ASTNode.py` maybe change this as well?
Or I didn't fetch the newest file when updating it ....

Comment (Raymond-yao) [reply README.md]: Nah loool that one is not updated either

**Pull request merged**